docs/source/architecture/slashing.rst

```diff
@@ -19,23 +19,43 @@ TBD (https://github.com/nucypher/nucypher/issues/803)
 How slashing affects stake
 --------------------------
 
-The goal of slashing is to reduce the number of tokens that belongs to a staking offender.
-In this case, the main task is not to violate the logic of locking tokens.
-The entire stake consists of:
+.. _`Sub-stakes`: https://docs.nucypher.com/en/latest/architecture/sub_stakes.html
 
-    * tokens which the staker can withdraw at any moment
+The goal of slashing is to reduce the number of tokens that belongs to a staking offender. While the reduction of tokens is intended to be punitive, once the tokens have been slashed, the slashing algorithm attempts to preserve the most efficient use of the offenders' remaining tokens based on their existing `Sub-stakes`_. The main task is not to violate the logic of locking tokens.
```

Comment:
"The main task is not to violate the logic of locking tokens."

Can you explain what you mean by this?

Reply:
we have such structure for staker in `StakingEscrow`
```
struct StakerInfo {
        uint256 value;
...
        SubStakeInfo[] subStakes;
    }
```
Technically slashing is just deacresing `value` field by slashing amount, because this `value` means how many tokens belong to staker. But what we call "slashing" here is to make all data still consistent.
If we just deacrese `value` and untouch `subStakes` array it could lead to such situation: sum if sub-stakes in some period in a future will be more than `value` and it means when staker will mint reward for that period he will get more tokens than he desirves. Also could be some other potential problems in the code but that is the main. So our task is to adjust values in  `subStakes` array

Reply:
What about?

```suggestion
Slashing aims to reduce the overall number of tokens that belongs to a staking offender. While the reduction of tokens is intended to be punitive, once the tokens have been slashed, the slashing algorithm attempts to preserve the most efficient use of the offenders' remaining tokens based on their existing `Sub-stakes`_. The main goal is to decrease the tokens belonging to the staker while considering how tokens are allocated across different sub-stakes.
```

Reply:
I like it, what do you think @mswilkison ?

Reply:
👍

Reply:
✔️